core/services/keeper/models_test.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+func TestUpkeepIdentifer_String(t *testing.T) {
+	for _, test := range []struct {
+		name string
+		id   string
+		hex  string
+	}{
+		{"small", "10", "UPx000000000000000000000000000000000000000000000000000000000000000a"},
+		{"large", "1000000000", "UPx000000000000000000000000000000000000000000000000000000003b9aca00"},
+		{"big", "5032485723458348569331745", "UPx0000000000000000000000000000000000000000000429ab990419450db80821"},
```

Comment:
👍

Reply:
These look very ugly right now :p but in production the upkeep IDs should be big integers